internal/integrationtests/integrationtests.go

```diff
@@ -74,74 +80,157 @@ type IntegrationTests struct {
+		if innerErr != nil {
+			return fmt.Errorf("building transaction string: %w", innerErr)
 		}
 		log.Ctx(ctx).Infof("feeBumpedTx[%d]: %s", i, txString)
 
 		it.assertFeeBumpTransactionResult(ctx, types.CreateFeeBumpTransactionRequest{Transaction: txXDR}, *feeBumpTxResponse)
 	}
 
-	// TODO: submitTx")
-	// TODO: waitForTxToBeInLedger")
+	// Step 5: wait for RPC to be healthy
+	fmt.Println("")
+	log.Ctx(ctx).Info("===> 5️⃣ [RPC] Waiting for RPC service to become healthy...")
+	err = WaitForRPCHealthAndRun(ctx, it.RPCService, 40*time.Second, nil)
+	if err != nil {
+		return fmt.Errorf("waiting for RPC service to become healthy: %w", err)
+	}
+
+	// Step 6: submit transactions to RPC
+	fmt.Println("")
+	log.Ctx(ctx).Info("===> 6️⃣ [RPC] Submitting transactions...")
+	hashes := make([]string, len(feeBumpedTxs))
+	for i, txXDR := range feeBumpedTxs {
+		res, err := it.RPCService.SendTransaction(txXDR)
+		if err != nil {
+			return fmt.Errorf("sending transaction %d: %w", i, err)
+		}
+		log.Ctx(ctx).Infof("✅ submittedTx[%d]: %+v", i, res)
+		if res.Status != entities.PendingStatus {
+			return fmt.Errorf("transaction %d failed with status %s and errorResultXdr %s", i, res.Status, res.ErrorResultXDR)
+		}
+		hashes[i] = res.Hash
+	}
+
+	// Step 7: poll the network for the transaction
+	fmt.Println("")
+	log.Ctx(ctx).Info("===> 7️⃣ [RPC] Waiting for transaction confirmation...")
+	retryOptions := []retry.Option{retry.Attempts(5), retry.Delay(6 * time.Second)}
+	for _, hash := range hashes {
+		err := it.waitForTransactionConfirmation(ctx, hash, retryOptions...)
+		if err != nil {
+			return fmt.Errorf("waiting for transaction confirmation: %w", err)
+		}
+		log.Ctx(ctx).Infof("✅ transaction %s confirmed on Stellar network", hash)
+	}
+
 	// TODO: verifyTxResult in wallet-backend")
 
 	return nil
 }
 
+func (it *IntegrationTests) waitForTransactionConfirmation(ctx context.Context, hash string, retryOptions ...retry.Option) error {
+	attemptsCount := 0
+	outerErr := retry.Do(
+		func() error {
+			attemptsCount++
+			log.Ctx(ctx).Infof("🔁 attemptsCount: %d", attemptsCount)
+			txResult, err := it.RPCService.GetTransaction(hash)
+			if err != nil {
+				return fmt.Errorf("getting transaction with hash %q: %w", hash, err)
+			}
+
+			switch txResult.Status {
+			case entities.NotFoundStatus:
+				return fmt.Errorf("transaction not found")
+			case entities.SuccessStatus:
+				return nil
+			case entities.FailedStatus:
+				err = fmt.Errorf("transaction with hash %q failed with status %s and errorResultXdr %s", hash, txResult.Status, txResult.ErrorResultXDR)
+				return retry.Unrecoverable(err)
+			default:
+				return fmt.Errorf("unexpected transaction status: %s", txResult.Status)
+			}
+		},
+		append(
+			retryOptions,
+			retry.Context(ctx),
+			retry.LastErrorOnly(true),
+		)...,
+	)
+
+	if outerErr != nil {
+		return fmt.Errorf("failed to get transaction status after %d attempts: %w", attemptsCount, outerErr)
```

Comment:
I see we have hardcoded the attempts here - are we sure to get the transaction confirmation in those attempts? What if it takes 6 attempts instead? Should we keep this value higher to ensure we can safely get the confirmation from RPC

Reply:
Good point! I updated the code to calculate the `retryCount` and `retryAfter` based on the transaction timebounds: https://github.com/stellar/wallet-backend/pull/160/commits/dcc6e2bce15257bd6f3469699552d049c9d82d13

Reply:
Ohh nice idea. Looks good to me